### PR TITLE
Search measure error fixes

### PR DESCRIPTION
**Pull request by myxrome**
fix error when duty_expression is not present for measure_component.
fixed explicit end_date in measure search.

### CODE REVIEW

**Comment by myxrome** [comment]
[[DIT] Search function not throwing accurate results.](https://trello.com/c/nCgLZg2i/447-dit-search-function-not-throwing-accurate-results)

**Comment by myxrome** [comment]
[[DIT] Search: end date filters for before/after should also include undefined (3)](https://trello.com/c/ksibeV56/372-dit-search-end-date-filters-for-before-after-should-also-include-undefined-3)

**Comment by myxrome** [comment]
[[DIT] Search is returning with zero results for an existing commodity code which has got measure associated with it (5)](https://trello.com/c/3ij50oX1/430-dit-search-is-returning-with-zero-results-for-an-existing-commodity-code-which-has-got-measure-associated-with-it-5)

**Comment by myxrome** [comment]
[[DIT] Change measures search to be able to search by multiple measure IDs - comma separated (2)](https://trello.com/c/t6VeX7hQ/506-dit-change-measures-search-to-be-able-to-search-by-multiple-measure-ids-comma-separated-2)

File: app/searches/measures/search_filters/date_universal.rb [hunk 1]
```diff
@@ -106,10 +113,11 @@ def is_after_clause
 
         def compare_sql(compare_operator)
           res = "#{field_name}::date #{compare_operator} ?"
-          res += " OR #{is_not_specified_clause}" if field_name == "validity_end_date"
+          res += " OR #{is_not_specified_clause}" if field_name == "validity_end_date" && OPERATORS_ALLOW_NIL.include?(operator)
 
           res
         end
+
```
Comment (matthewford) [inline app/searches/measures/search_filters/date_universal.rb]: not sure if we need this new line

Comment (myxrome) [reply app/searches/measures/search_filters/date_universal.rb]: ok, I'll remove it with next commit.

**Pull request merged**